src/dbl_linked_list.py

```diff
@@ -0,0 +1,125 @@
+                self.tail = nodes[-1]
+            except TypeError:
+                self.push(maybe_an_iterable)
+
+    def push(self, value):
+        """Add a node at the beginning of the line."""
+        self.head = Dbl_Node(value, self.head)
+        self._size += 1
+        if self._size == 1:
+            self.tail = self.head
+        elif self._size == 2:
+            nodes = [node for node in self._iterate_from(self.head)]
+            self.tail = nodes[-1]
+            nodes[-1].prev = self.head
+        else:
+            self.head.nxt.prev = self.head
+
+    def pop(self):
+        """Pop the first value off the head of LL and return it."""
+        if self.head is None:
+            raise IndexError("Cannot pop from an empty Linked List.")
+        val = self.head.value
+        self.head = self.head.nxt
+        self.head.prev = None
+        self._size -= 1
+        return val
+
+    def shift(self):
+        """will remove the last value from the tail of the list and return it."""
+        if self.tail is None:
+            raise IndexError("Cannot shift (remove the last node in the line) from an empty Linked List.")
+        val = self.tail.value
+        self.tail.prev.nxt = None
+        self.tail = self.tail.prev
+        self._size -= 1
+        return val
+
+    def append(self, value):
+        """will append a new node at the end of the line and will set the new tail to be this node."""
+        new_node = Dbl_Node(value, None, self.tail)
+        nodes = [node for node in self._iterate_from(self.head)]
+        if nodes:
+            self.tail = new_node
+            nodes[-1].nxt = self.tail
+            self._size += 1
+        else:
+            self.head = new_node
+            self.tail = new_node
+            self._size += 1
+
+    def remove(self, value):
+        """Remove the given node from the LL."""
+        nodes = [node for node in self._iterate_from(self.head)]
+        r_node = [node for node in self._iterate_from(self.head) if node.value == value]
+        if not r_node:
+            raise ValueError("ERROR: That node is not in this linked list.")
+        for node in self._iterate_from(self.head):
+            if r_node[0].value == node.value:
+                if self.head.value == r_node[0].value:
+                    self.pop()
+                    return "Succesfully removed Node with value: {0}. New head set to {1}".format(r_node[0].value, self.head.value)
+                elif node.nxt is None:
+                    self.shift()
+                    return "Succesfully removed Node with value: {} using Shift()".format(r_node[0].value)
+                else:
+                    r_node[0].prev.nxt = r_node[0].nxt
+                    r_node[0].nxt.prev = r_node[0].prev
+                    self._size -= 1
+                    return "Succesfully removed Node with value: {}".format(r_node[0].value)
+
+    def _iterate_from(self, list_item):
```

Comment:
Nice take on making your DLL iterable! in order for this to hook into the `for` operator in Python, this function should be named `__iter__`. I think so anyway; I might be wrong. regardless, check out the docs.